scylla/src/transport/cluster.rs

```diff
@@ -381,25 +381,6 @@ impl ClusterData {
         &self.keyspaces
     }
 
-    /// Access datacenter details collected by the driver
-    /// Returned `HashMap` is indexed by names of datacenters
-    pub fn get_datacenters_info(&self) -> HashMap<String, Datacenter> {
-        self.locator
-            .datacenter_names()
-            .iter()
-            .map(|dc_name| {
-                let nodes = self
-                    .locator
-                    .unique_nodes_in_datacenter_ring(dc_name)
-                    .unwrap()
-                    .to_vec();
-                let rack_count = nodes.iter().map(|node| node.rack.as_ref()).unique().count();
-
-                (dc_name.clone(), Datacenter { nodes, rack_count })
-            })
-            .collect()
-    }
-
     /// Access details about nodes known to the driver
```

Comment:
This is part of the "cluster examination" API. I agree with removing this particular method, because it's extremely wasteful, but I think you should include something similar (in the aspect of being a convenient way to examine the cluster) in the metadata refactor.

Reply:
I will think of a proper API and implementation during refactor.

Please note that this doesn't remove cluster examination API, only 1 (imo not very useful) method of it. The "primary" method, `get_nodes_info`, is still present.